src/Swift.Runtime/src/Swift/SwiftArray.cs

```diff
@@ -150,80 +161,58 @@ public unsafe int Count
     /// </summary>
     public unsafe void Append(Element item)
     {
-        IntPtr payload = IntPtr.Zero;
         var metadata = SwiftObjectHelper<SwiftArray<Element>>.GetTypeMetadata();
-        try
-        {
-            payload = (IntPtr)NativeMemory.Alloc(ElementSize);
-            SwiftMarshal.MarshalToSwift(item, payload);
-
-            fixed (void* bufferPtr = &_buffer)
-            {
-                SwiftArrayPInvokes.Append(payload, metadata, new SwiftSelf(bufferPtr));
-            }
-        }
-        finally
-        {
-            NativeMemory.Free((void*)payload);
-        }
+        bool _success = false;
+        _payload.DangerousAddRef(ref _success);
+        byte* payload = stackalloc byte[(int)ElementSize];
+        Span<byte> span = new Span<byte>(payload, (int)ElementSize);
+        SwiftMarshal.MarshalToSwift(item, span);
+        SwiftArrayPInvokes.Append((IntPtr)payload, metadata, new SwiftSelf((void*)_payload.Handle));
```

Comment:
Where is `item` being used here? Is this related to `payload`?

Reply:
I think this would be clearer if `ref span` was used.

Reply:
Instead of the `Span<>` being passed in and magically updated, it might be clearer to have it pass it by `ref`.

```csharp
SwiftMarshal.MarshalToSwift(item, ref span);
```

If this is too much of a change or doesn't provide a lot of clarity, then you can keep it as is.

Reply:
I want to decouple memory allocation from the marshalling logic. `MarshalToSwift` is only used for passing generic parameters at PInvoke boundary, and the memory is allocated with `stackalloc`. The call to `MarshalToSwift` only writes to the span’s underlying memory, it doesn’t reassign or replace the span itself. 

I changed the implementation for clarity to first allocate the span using `stackalloc` and then take the pointer.

Reply:
Thanks for the explanation, that makes sense. I hadn’t considered that allowing the marshalling system to modify the required memory could be beneficial. I’ve updated the parameter to be passed by ref for clarity and to support potential future changes.